src/app/datepicker/datepicker.directive.ts

```diff
@@ -0,0 +1,200 @@
+    }
+  }
+
+  initHandlers() {
+    this.handlers = {
+      label: () => this.handleLabel(),
+      placeholder: () => this.handlePlaceholder(),
+    };
+  }
+
+  initElements() {
+    this.inputContainerElement = $(this.elementRef.nativeElement).parent('.input-field');
+    this.inputElement = $(this.elementRef.nativeElement);
+    this.labelElement = this.createLabelElement();
+  }
+
+  initDatepicker() {
+    // set default format/formatSubmit options
+    if (this.format) {
+      this.options.format = this.format;
+    }
+    if (this.formatSubmit) {
+      this.options.formatSubmit = this.formatSubmit
+    }
+
+    // extends onClose function to fix datepicker focus issue
+    // https://github.com/Dogfalo/materialize/issues/2067#issuecomment-142107599
+    const onCloseFn = this.options && this.options.onClose || (() => {});
+    this.options = Object.assign({}, this.options, {
+      onClose: (event) => {
+        onCloseFn(event);
+        this.renderer.invokeElementMethod(document.activeElement, 'blur');
+      },
+    });
+
+    this.renderer.invokeElementMethod(this.inputElement, 'pickadate', [this.options]);
+
+    if (this.ngControl) {
+      // set datepicker initial value according to ngControl
+      this.picker.set('select', this.ngControlValue, { format: this.formatSubmit });
+
+      // set ngControl value according to selected date in datepicker
+      this.picker.on('set', () => {
+        // handle stop propagation
+        if (this.stopChangePropagation) {
+          this.stopChangePropagation = false;
+          return;
+        } else {
+          this.stopChangePropagation = true;
+        }
+
+        // apply options.formatSubmit to ngControl value
+        const submitValue = this.formatSubmit
+          ? this.picker.get('select', this.formatSubmit)
+          : this.picker.get('value');
+        this.ngControl.control.setValue(submitValue);
+
+        // apply options.format to input text
+        const formatValue = this.format
+          ? this.picker.get('select', this.format)
+          : this.picker.get('value');
+        this.inputElement.val(formatValue);
+
+        // set label active status
+        this.setLabelActive();
+      });
+    }
+  }
+
+  initInputSubscriber() {
+    if (this.ngControl) {
+      this.inputValueSubscription = this.ngControl.valueChanges.subscribe(() => {
+        // handle stop propagation
+        if (this.stopChangePropagation) {
+          this.stopChangePropagation = false;
+          return;
+        } else {
+          this.stopChangePropagation = true;
+        }
+
+        // set selected date in datepicker according to ngControl value
+        this.picker.set('select', this.ngControlValue, { format: this.formatSubmit });
+
+        // set label active status
+        this.setLabelActive();
+      });
+    }
+  }
+
+  createLabelElement() {
+    const labelElement = document.createElement('label');
+    labelElement.setAttribute('for', this.id);
```

Comment:
Did you try to create a datepicker without date?

Reply:
Without date yes and if you meant without label ... the test you commented below covers it